django/db/models/query.py

```diff
@@ -1545,7 +1545,12 @@ def get_prefetcher(instance, through_attr, to_attr):
                 if hasattr(rel_obj, 'get_prefetch_queryset'):
                     prefetcher = rel_obj
                 if through_attr != to_attr:
-                    is_fetched = hasattr(instance, to_attr)
+                    # Special case cached_property instances as using hasattr
```

Comment:
I'm a bit tired right now but I had trouble understanding this. Maybe "as using" -> "because" would be clearer.